### DISCUSSION TITLE
feature request - fill hole method

### DISCUSSION THREAD

**Discussion by yetigit** [elalish/manifold#513] [Ideas]
At the moment I'm dealing with plenty of opened meshes; though I could do a rudimentary hole filling function myself and use it before submitting any meshes to Manifold or I could use a third party lib to do so, it would still be nice to see a function like that in the Manifold lib.

**Comment by yetigit** [comment]
using openvdb/voxel like trick is out of my options since I need to retain the original mesh likeness and cannot afford the extra density

**Comment by elalish** [comment]
Hole filling is *extremely* arbitrary and heuristic, which puts it outside the scope of what Manifold is about (precision and robustness). The best algorithms I saw for this were in MeshMixer. It's an Autodesk product now, but it was once open source - some of those algorithms may still be available. It would be great to have a nice open-source mesh cleanup library, but I don't think I want to maintain one.
